### PR TITLE
clips: 6.30 -> 6.31; supersedes and closes #109914

### PR DESCRIPTION
**Pull request by league**
###### Motivation for this change

New stable release. This also follows a bot-initiated update #109914 that didn't work out on its own, but it's now specified to (hopefully) be more amenable to auto-updates in the future.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114132` at 4236771f run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/6e0e76c1a0f591b7398f5a6162ac6a17">clips</a></li></li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/development/interpreters/clips/default.nix:12:3:
```
   |
12 |   buildPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/interpreters/clips/default.nix:15:3:
```
   |
15 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/interpreters/clips/default.nix [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
-  src = fetchurl {
-    url = "mirror://sourceforge/clipsrules/CLIPS/6.30/clips_core_source_630.tar.Z";
-    sha256 = "1r0m59l3mk9cwzq3nmyr5qxrlkzp3njls4hfv8ml85dmqh7n3ysy";
+  src = let v = builtins.replaceStrings [ "." ] [ "" ] version;
+  in fetchurl {
+    url =
+      "mirror://sourceforge/clipsrules/CLIPS/${version}/clips_core_source_${v}.tar.gz";
+    sha256 = "165k0z7dsv04q432sanmw0jxmxwf56cnhsdfw5ffjqxd3lzkjnv6";
   };
   buildPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/clips/default.nix:12]:
```suggestion
```

File: pkgs/development/interpreters/clips/default.nix [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
-    make -C core -f ../makefiles/makefile.gcc
+    runHook preBuild
+    make -C core
+    runHook postBuild
   '';
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/clips/default.nix:16]:
```suggestion
  makeFlags = [ "-C" "core" ];
```

File: pkgs/development/interpreters/clips/default.nix [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
+  version = "6.31";
   pname = "clips";
-  src = fetchurl {
-    url = "mirror://sourceforge/clipsrules/CLIPS/6.30/clips_core_source_630.tar.Z";
-    sha256 = "1r0m59l3mk9cwzq3nmyr5qxrlkzp3njls4hfv8ml85dmqh7n3ysy";
+  src = let v = builtins.replaceStrings [ "." ] [ "" ] version;
+  in fetchurl {
+    url =
+      "mirror://sourceforge/clipsrules/CLIPS/${version}/clips_core_source_${v}.tar.gz";
+    sha256 = "165k0z7dsv04q432sanmw0jxmxwf56cnhsdfw5ffjqxd3lzkjnv6";
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/clips/default.nix:10]:
```suggestion
  in fetchurl {
    url = "mirror://sourceforge/clipsrules/CLIPS/${version}/clips_core_source_${builtins.replaceStrings [ "." ] [ "" ] version}.tar.gz";
    sha256 = "165k0z7dsv04q432sanmw0jxmxwf56cnhsdfw5ffjqxd3lzkjnv6";
```

**Review by SuperSandro2000 — APPROVED** [review]
other than the little simplification LGTM

**Comment by league** [comment]
Thanks, should be ready to merge once checks finish.

**Pull request merged**